primer/brand
# `Hero` component refactor

## Summary



Provides the existing `Hero` component with a composable API. 

- [`Hero` documentation](https://primer-cfabcb7d9c-26139705.drafts.github.io/components/Hero)
- [`Hero` storybook](https://primer-cfabcb7d9c-26139705.drafts.github.io/storybook/?path=/story/components-hero--default) 

## List of notable changes:



- Refactors `Hero` to use composable API.

## What should reviewers focus on?

- The ability to extend `Hero` components (i.e. `Hero.Heading`, `Hero.Description`, etc).
- No visual changes being introduced.

## Steps to test:



1. Open `Hero` component in Storybook instance provided by this PR
2. Compare against existing `Hero` in production, there should be **no** visual change


## Supporting resources (related issues, external links, etc):

- https://github.com/github/primer/issues/2064
- https://github.com/primer/brand/issues/63
- [API exploration](https://www.figma.com/file/8PgBjRPhhF7IGs4ZMCwUnH/Primer-Brand---Hero-refactor-API?node-id=0-1&t=wvuayIEaVHRYFMPO-0)

## Contributor checklist:


## Reviewer checklist:
